lib/src/components/NxList/NxList.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, useRef } from 'react';
+import classnames from 'classnames';
+import useMergedRef from '@react-hook/merged-ref';
+import withClass from '../../util/withClass';
+
+import NxLoadingSpinner from '../NxLoadingSpinner/NxLoadingSpinner';
+import NxLoadError from '../NxLoadError/NxLoadError';
+
+import NxListText from './NxListText';
+import NxListSubtext from './NxListSubtext';
+import NxListActions from './NxListActions';
+import NxListButtonItem from './NxListButtonItem';
+import NxListLinkItem from './NxListLinkItem';
+import NxListItem from './NxListItem';
+import { NxListProps, NxListComponent, nxListPropTypes } from './types';
+import useEmptyComponent from '../../util/useEmptyComponent';
+
+
+const NxList = forwardRef<HTMLUListElement, NxListProps>((props: NxListProps, externalRef) => {
```

Comment:
To make sure the function gets named correctly for debugging purposes
```suggestion
const NxList = forwardRef<HTMLUListElement, NxListProps>(function NxList(props: NxListProps, externalRef) {
```